librosa/core/spectrum.py

```diff
@@ -1335,12 +1332,22 @@ def iirt(y, sr=22050, win_length=2048, hop_length=None, center=True,
             cur_filter_output = scipy.signal.sosfiltfilt(cur_filter,
                                                          y_resampled[cur_sr_idx])
 
-        # frame the current filter output
-        cur_frames = util.frame(np.ascontiguousarray(cur_filter_output),
-                                frame_length=win_length_STMSP,
-                                hop_length=hop_length_STMSP)
+        factor = float(sr) / float(cur_sr)
+        hop_length_STMSP = hop_length / factor
+        win_length_STMSP_round = int(round(win_length / factor))
```

Comment:
Can these be written with integer division (`//`)?

The float casts for `factor` calculation are no longer necessary, since we don't support python2.

Reply:
I removed the float casts. I don't see an obvious application of integer division without changing the behavior. `win_length_STMSP_round` uses rounding, so we could convert it like `(win_length + factor // 2) // factor` (see [SO](https://stackoverflow.com/questions/3950372/round-with-integer-division)), but that is less readable IMO.

Reply:
Yeah, that's fine for win length, but hop length should be integer division.

The current implementation hacks around this by casting the arange to int before doing anything with it, but that seems like it could introduce unpredictable bugs.  Better to make sure we have the right types up front.

Reply:
The idea of the fix is that the hop size is passed as non-integer to arange. Doing so, we avoid the accumulated rounding errors. The discretization only happens at the very end, when indices are used. Why do you see the introduction of unpredictable bugs?

Reply:
> The idea of the fix is that the hop size is passed as non-integer to arange.

Perhaps I misunderstood then.



> The discretization only happens at the very end, when indices are used. Why do you see the introduction of unpredictable bugs?

Force-casting after the fact can hide mistakes that we could otherwise catch, so as a general rule, I prefer to make the calculations explicit up front to avoid silent errors.  But if this behavior is intentional and part of the fix, I'm fine with it.  (Maybe add a line of documentation explaining what's going on though.)